common/src/main/scala/com/aliyun/emr/rss/common/protocol/message/ControlMessages.scala

```diff
@@ -182,12 +182,14 @@ sealed trait Message extends Serializable{
       case GetReducerFileGroupResponse(status, fileGroup, attempts) =>
         val builder = TransportMessages.PbGetReducerFileGroupResponse.newBuilder()
           .setStatus(status.getValue)
-        if (fileGroup != null) {
-          builder.addAllFileGroup(fileGroup.map(arr => PbFileGroup.newBuilder()
-            .addAllLocaltions(arr.map(PartitionLocation.toPbPartitionLocation(_)).toIterable.asJava)
-            .build()).toIterable.asJava)
+        if (!fileGroup.isEmpty) {
+          builder.addAllFileGroup(fileGroup.map { arr =>
+            PbFileGroup.newBuilder()
+              .addAllLocaltions(arr.map(PartitionLocation.toPbPartitionLocation).toIterable.asJava)
+              .build()
+          }.toIterable.asJava)
         }
-        if (attempts != null) {
+        if (!attempts.isEmpty) {
```

Comment:
There is no need to check whether the list has elements or not. The empty list will skip the map operation.

Reply:
DOne